File: winforms/aaudiocore/Strategies/NormalStrategy.cs
using AAudioCore.Contracts;
using AAudioCore.Models;

namespace AAudioCore.Strategies;

public sealed class NormalStrategy : IPlaybackStrategy
{
    public string Name => "normal";

    public int SelectNext(int currentIndex, IReadOnlyList<PlaybackTrack> queue)
    {
        if (queue.Count == 0) return -1;
        return currentIndex >= queue.Count - 1 ? 0 : currentIndex + 1;
    }
}

File: react/aaudiocore/src/strategies/normal-strategy.ts
import type { IPlaybackStrategy } from "../contracts/playback-strategy";
import type { PlaybackTrack } from "../models/track";

export class NormalStrategy implements IPlaybackStrategy {
  readonly name = "normal";

  selectNext(currentIndex: number, queue: PlaybackTrack[]): number {
    if (queue.length === 0) return -1;
    return currentIndex >= queue.length - 1 ? 0 : currentIndex + 1;
  }
}
